src/rgw/rgw_http_client_curl.cc

```diff
@@ -0,0 +1,121 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+extern "C" {
+#include <curl/curl.h>
+}
```

Comment:
this still builds without the extern "C", right? i know it just moved from rgw_main.cc, but i'd like to drop it if isn't not needed